homeassistant/components/tradfri/base_class.py

```diff
@@ -34,12 +40,14 @@ class TradfriBaseClass(Entity):
 
     _attr_should_poll = False
 
-    def __init__(self, device, api, gateway_id):
+    def __init__(
+        self, device: Command, api: Callable[[str], Any], gateway_id: str
+    ) -> None:
         """Initialize a device."""
         self._api = handle_error(api)
-        self._device = None
-        self._device_control = None
-        self._device_data = None
+        self._device: Command | None = None
+        self._device_control: SocketControl = None
+        self._device_data: Socket = None
```

Comment:
For clarification, that `None` is allowed here, the typing should be `SocketControl | None` and `Socket | None`.

Reply:
This is one of the more tricky ones. Let me try to explain

If you do:
```
    self._device_control: SocketControl | None = None
```
you can assign None in the running code, but gives problem with e.g.:
```
    jan = self._device_control.status()
```
where mypy will complain that it can be None, and this illegal.

Using it as I do permits having the primary value None, but not assigning to None, and thus no problem with the jan = statement. I think mypy checks that the code do not refer to self._device_control before it is assigned.

Reply:
Normally we type it as `None` if we initialize the name to `None` to force us to explicitly check or assert for the type before we use the name, to avoid bugs. We have some exceptions in the core, eg `hass` on the Entity which we don't type as `None` to avoid have to check for that in each integration, since the core handles this.

Reply:
That is a very valid argument, I just wanted to safe a couple of “if self._device_control is None”, I will change that.

Apart from that just saw that the mypy definition have changed so I need to do a couple of other changes as well.